numpy/testing/_private/utils.py

```diff
@@ -388,6 +388,12 @@ def assert_equal(actual, desired, err_msg='', verbose=True):
 
     except (TypeError, ValueError, NotImplementedError):
         pass
+    except DeprecationWarning:
+        # version 1.18
+        # gisnan used to raise for datetime64/timedelta64 (dt64) since they had
+        # no isnan implementation. That call no longer raises. However, a
+        # DeprecationWarning is raised when comparing dt64 to 0.
```

Comment:
[this comment](https://github.com/numpy/numpy/pull/14841#discussion_r344258043) still applies - this will still emit noisy warnings if I don't have warnings set to raise - we should either:
* Suppress warnings, rather than catching them
* Locally promote warnings to raise
* Special case datetime types to avoid this path
* Call `cancast` somehow to avoid this path

Reply:
special casing seems like the easiest